### PR TITLE
feat: Add copy code functionality to Color Picker

### LINKED ISSUE #497: Add a copy function for ColorPicker
I made a short video presentation:   https://youtu.be/_rfFx7NeQQE
After the color picker selects the color, click the copy button can copy the Java code or CSS code of the color

### Expected Behavior

After the color picker selects the color, click the copy button, and we can get the CSS code or Java code of the current color;

### Current Behavior

For example, gradient can only generate corresponding fxml code We can't generate the Java code or CSS code .

### Context

    Many times, we need color's java code or CSS code; For example, make cool gradient animation with color; 
    Or we need to modify the color of control through CSS;
    It is troublesome to write the color code manually, especially the gradient color;
    I have developed an idea plug-in with about 1000 downloads per month,
    Because IDEA supports swing development plug-ins, its support for JavaFX is not good enough, So I want to migrate the function of copying color codes to scenebuilder,
my idea plugin:   https://plugins.jetbrains.com/plugin/17514-javafx-tools
    In the future, I plan to add other very useful functions to scenebuilder.

### LINKED ISSUE #503: CSS Error parsing Warning may be thrown
Control: Color Picker. 
CSS warning may be thrown when the value of the alpha is a decimals stored in scientific notation;

### Expected Behavior

Alpha values should be rounded.

### Current Behavior

For example, alpha value is 7.552884496591084E-4
e.g.  :  The console will output: 
>  CSS Error parsing '*{-fx-fill: hsb(224.0,100.0%,92.0%,7.552884496591084E-4)}: Unexpected token '7.552884496591084E' at [1,35]

### Steps to Reproduce

Write an app, use colorpicker, drag the slider several times to adjust the value of alpha, 
and usually throw this warning when the transparency is close to 0

### Your Environment

Win10 x64;    JDK17

### Screenshots

Location: 
https://github.com/gluonhq/scenebuilder/blob/master/kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/colorpicker/ColorPicker.java
Line 462
Possible solution:
Source code: sb.append(alpha);
Modified code: sb.append(alpha_rounded); //alpha_rounded already exists

### PR DESCRIPTION
**Pull request by leewyatt**
Signed-off-by: leewyatt <leewyatt7788@gmail.com>

https://twitter.com/LeeWyatt_7788/status/1485836423446278145
Add copy code function to color selector;
Because many times we need java code or CSS code for colors, especially gradient colors;

### Issue

Fixes #497 
Fixes #503 

### Progress

I have finished this function

- [x] Change must not contain extraneous whitespace
- [x] License header year is updated, if required
- [x] Verify the contributor has signed [Gluon Individual Contributor License Agreement (CLA)](https://docs.google.com/forms/d/16aoFTmzs8lZTfiyrEm8YgMqMYaGQl0J8wA0VJE2LCCY)

### CODE REVIEW

**Comment by abhinayagarwal** [comment]
Hi @leewyatt ,

Warm welcome to Scene Builder family 👋 

Thank you for your contribution. Before we proceed with reviewing the changes, please log an issue and describe how the enhancement would be useful for Scene Builder. Then link the issue to the PR by updating the 'Fixes #<issue-number>'.

**Comment by leewyatt** [comment]
@abhinayagarwal  Thank you for your advice~;   Fixes #497

**Comment by leewyatt** [comment]
It's too late. I want to go to bed. I'll resubmit it tomorrow.

**Comment by abhinayagarwal** [comment]
The PR is currently failing a [Semantic PR title](https://github.com/gluonhq/scenebuilder/issues/451#issuecomment-1019902968) check. Please update the title to `feat: Add copy code function to Color Picker `.

**Comment by leewyatt** [comment]
@abhinayagarwal Thanks for your help. I seem to be having trouble with commits. I may need to talk to a friend who has experience with commits.

**Comment by abhinayagarwal** [comment]
@leewyatt I have made the necessary change to the PR title. It is now passing the check :)

If the PR is WIP, you can mark it as 'draft': https://github.blog/2019-02-14-introducing-draft-pull-requests/

**Comment by leewyatt** [comment]
wow! Thank you so much.

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java
Comment (abhinayagarwal) [inline kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java:1]: Add license header to the file

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java
Comment (abhinayagarwal) [inline kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java:14]: Please add some tests to confirm the validity of this method.

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java
Comment (abhinayagarwal) [inline kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java:44]: Please add some tests to confirm the validity of this method.

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java
Comment (abhinayagarwal) [inline kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java:3]: Use single imports

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/colorpicker/ColorPicker.java
Comment (abhinayagarwal) [inline kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/colorpicker/ColorPicker.java:44]: Use single import

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/colorpicker/ColorPicker.java
Comment (abhinayagarwal) [inline kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/colorpicker/ColorPicker.java:34]: Update license header year for this class

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java
Comment (abhinayagarwal) [inline kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java:28]: Is it possible to create a Paint object which is an instance of either `LinearGradient` or `RadialGradient` from the Scene Builder's ColorPicker dialog?

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/colorpicker/ColorPicker.java
Comment (leewyatt) [reply kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/colorpicker/ColorPicker.java:34]: image
Hi, thanks for the suggestion; should I modify it the way it is in the picture.

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java
Comment (leewyatt) [reply kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java:28]: There is a PaintPickerController object in ColorPicker, we can get the current Paint (Color or LinearGradient or RadialGradient) of ColorPicker through PaintPickerController

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/colorpicker/ColorPicker.java
Comment (leewyatt) [reply kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/colorpicker/ColorPicker.java]: Thanks, it has been edited

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java
Comment (leewyatt) [reply kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java]: Thanks, it has been edited

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java
Comment (leewyatt) [reply kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java:80]: Thanks, the test class has been added;
com.oracle.javafx.scenebuilder.kit.util.PaintConvertUtilTest

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java
Comment (leewyatt) [reply kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java:50]: Thanks, the test class has been added;
com.oracle.javafx.scenebuilder.kit.util.PaintConvertUtilTest

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java
Comment (leewyatt) [reply kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java:32]: thanks, the license has been added

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/colorpicker/ColorPicker.java
Comment (abhinayagarwal) [reply kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/colorpicker/ColorPicker.java:35]: Please find the exact license header below:

```
/*
 * Copyright (c) 2022, Gluon and/or its affiliates.
 * All rights reserved. Use is subject to license terms.
 *
 * This file is available and licensed under the following license:
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 *
 *  - Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 *  - Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in
 *    the documentation and/or other materials provided with the distribution.
 *  - Neither the name of Oracle Corporation and Gluon nor the names of its
 *    contributors may be used to endorse or promote products derived
 *    from this software without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
 * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
 * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
 * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
 * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
 * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
 * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
 * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 ```

Comment (leewyatt) [reply kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/colorpicker/ColorPicker.java:35]: Thank you for your suggestion, I noticed a little change in the content of the license, but I am a bit confused about the timing, I contacted you via twitter

File: kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java
Comment (AlmasB) [inline kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java:51]: We are slowly getting rid of null related code, so we should not be passing in null or returning null. Based on this, if `fxPaint` is null, then it is a call site issue, not ours. Given this is new API, it is not being called in the current codebase, so this check is redundant.

Comment (leewyatt) [reply kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtil.java]: Thank you! Yes paint is non-empty every time; this part of the judgment has been deleted

File: kit/src/test/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtilTest.java
Comment (leewyatt) [reply kit/src/test/java/com/oracle/javafx/scenebuilder/kit/util/PaintConvertUtilTest.java]: Thank you. I have removed the blank lines for this part

**Review by abhinayagarwal — APPROVED** [review]
Changes LGTM. Can merge once we have an approval from @AlmasB

**Comment by leewyatt** [comment]
> Changes LGTM. Can merge once we have an approval from @AlmasB

Thank you for always helping me.

**Review by AlmasB — APPROVED** [review]
Thanks, looks good

**Pull request merged**
